EpikCord/__init__.py

```diff
@@ -4186,4 +4188,90 @@ async def default_failure(self, interaction):
 class CommandUtils:
     
     def check(self, callback):
-        return Check(callback)
\ No newline at end of file
+        return Check(callback)
+
+class BaseConnectable:
+    def __init__(self,client:Client):
+        self.client = client
+        self.ws = None
+        self.IDENTIFY = 0
+        self.SELECT_PROTOCOL = 1
+        self.READY = 2
+        self.HEARTBEAT = 3
+        self.HELLO = 8
+        self.connected = False
+        self.server_set = False
+        self.state_set = False
+        self.sequence = None
+        self.endpoint:str = None
```

Comment:
Typing is incorrect:

```suggestion
        self.endpoint: Optional[str] = None
```